pages/registry/[[...page]].tsx

```diff
@@ -29,15 +35,32 @@ function RegistryLayout(props) {
   )
 }
 
-const { getStaticPaths, getStaticProps } = getStaticGenerationFunctions({
-  strategy: 'fs',
-  localContentDir: CONTENT_DIR,
-  navDataFile: NAV_DATA,
-  product: SOURCE_REPO,
-  githubFileUrl(filepath) {
-    return `https://github.com/hashicorp/${SOURCE_REPO}/blob/${DEFAULT_BRANCH}/${filepath}`
-  },
-})
-export { getStaticPaths, getStaticProps }
+const { getStaticPaths, getStaticProps } = getStaticGenerationFunctions(
+  process.env.IS_CONTENT_PREVIEW &&
+    process.env.PREVIEW_FROM_REPO === SOURCE_REPO
+    ? {
+        strategy: 'fs',
+        localContentDir: CONTENT_DIR,
+        navDataFile: NAV_DATA,
+        product: SOURCE_REPO,
+        githubFileUrl(filepath) {
+          // This path rewriting is meant for local preview from `terraform-docs-common`.
+          filepath = filepath.replace('preview/', '')
+          return `https://github.com/hashicorp/${SOURCE_REPO}/blob/${DEFAULT_BRANCH}/website/${filepath}`
+        },
+        remarkPlugins: (params) => [
+          remarkRewriteAssets({
+            product: SOURCE_REPO,
+            version: process.env.CURRENT_GIT_BRANCH,
+            getAssetPathParts: (nodeUrl) => ['website', nodeUrl],
+          }),
+        ],
+      }
+    : {
+        strategy: 'remote',
+        basePath: BASE_ROUTE,
+        product: SOURCE_REPO,
+      }
```

Comment:
Should we add `revalidate` here?

Reply:
Yes — I was thinking to add that to **all** the `[[...page]]` components in one fast-follow PR. Thoughts?